storage/pebble/registers.go

```diff
@@ -73,23 +114,45 @@ func (s *Registers) Store(
 			return fmt.Errorf("failed to set key: %w", err)
 		}
 	}
-
-	err := batch.Commit(pebble.Sync)
+	// increment height and commit
+	err := batch.Set(latestHeightKey(), EncodedUint64(height), nil)
+	if err != nil {
+		return fmt.Errorf("failed to update latest height %d", height)
+	}
+	err = batch.Commit(pebble.Sync)
 	if err != nil {
 		return fmt.Errorf("failed to commit batch: %w", err)
 	}
+	s.latestHeight = height
 
 	return nil
 }
 
-// TODO: Finish implementation after deciding prefixes and/or badger use
-
 // LatestHeight Gets the latest height of complete registers available
-func (s *Registers) LatestHeight() (uint64, error) {
-	panic("not implemented")
+func (s *Registers) LatestHeight() uint64 {
```

Comment:
Generally, I feel caching of the last value should be encapsulated within this function, I don't see much value in splitting it into two separate functions. Am I missing something?

Reply:
So instead of having two functions one to get from db and one to get the cached value I would just put a check inside this function to see if we already cached height and if so return it otherwise load it, cache it, and return it. Read through cache.